pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -95,22 +96,70 @@ func (ic *ifConfigurator) moveVFtoContainerNS(vfNetDevice string, containerID st
 	hostIface := result.Interfaces[0]
 	containerIface := result.Interfaces[1]
 
+	link, err := ic.netlink.LinkByName(vfNetDevice)
+	klog.V(2).Infof("Get link of vfNetDevice: %s, link: %+v", vfNetDevice, link)
+	if err != nil {
+		return fmt.Errorf("error getting VF link: %v", err)
+	}
+
+	// Rename the device in a temp NS to avoid race condition.
+	tempNS, err := tempNetNS()
+	if err != nil {
+		return fmt.Errorf("failed to create tempNS: %v", err)
+	}
+	defer tempNS.Close()
+
+	// Move the host VF device into tempNS
+	if err = ic.netlink.LinkSetNsFd(link, int(tempNS.Fd())); err != nil {
+		return fmt.Errorf("failed to move %q to tempNS: %v", link, err)
+	}
+
+	if err = tempNS.Do(func(hostNS ns.NetNS) error {
```

Comment:
No need to restore these changes after a failure?

Reply:
When there is a failure after `moveVFtoContainerNS` returns, do we need to restore the interface changes too? The question is more for the SR-IOV offload case (through `configureInterfacesCommon` -> `configureContainerSriovLinkOnBridge`).

Reply:
Hi @jianjuns  are you asking the changes like 'rename VF representor to hostIfaceName'? I suppose we should restore as well, but probably not in this PR. I noticed that Antonin mentioned we didn't verify ovs-offload feature for a long time. We may discuss if we'd like to remove it or improve it.

Reply:
If we do not recover the name for OVS offload VF, seems we do not have an issue here, but probably good to add a comment on `moveVFtoContainerNS`, saying if there is a failure in the caller after the func returns, the caller should restore the VF name, for the secondary network case.